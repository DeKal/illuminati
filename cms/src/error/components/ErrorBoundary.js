import React from 'react'
import PropTypes from 'prop-types'
import Error from 'error/components/Error'

class ErrorBoundary extends React.Component {
  constructor(props) {
    super(props)
    this.state = { hasError: false }
  }

  static getDerivedStateFromError(error) {
    return { error, hasError: true }
  }

  render() {
    if (this.state.hasError || this.props.reduxError) {
      // You can render any custom fallback UI
      return <Error error={this.props.reduxError || this.state.error} />
    }

    return this.props.children
  }
}
ErrorBoundary.propTypes = {
  children: PropTypes.any,
  reduxError: PropTypes.object
}

export default ErrorBoundary
