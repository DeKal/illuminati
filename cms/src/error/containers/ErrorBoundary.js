import { connect } from 'react-redux'
import ErrorBoundary from 'error/components/ErrorBoundary'
import { getAppError } from 'core/state/app/selectors'

const mapStateToProps = state => ({
  reduxError: getAppError(state)
})

export default connect(mapStateToProps)(ErrorBoundary)
