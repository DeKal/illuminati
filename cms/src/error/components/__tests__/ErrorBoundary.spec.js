import * as React from 'react'
import { shallow } from 'enzyme'
import ErrorBoundary from '../ErrorBoundary'

describe('ErrorBoundary', () => {
  it('should render children content correctly', function() {
    const wrap = shallow(
      <ErrorBoundary>
        <div>Content</div>
      </ErrorBoundary>
    )
    expect(wrap).toMatchSnapshot()
  })

  it('should render error correctly', function() {
    const wrap = shallow(
      <ErrorBoundary reduxError={{ msg: 'Error' }}>
        <div>Content</div>
      </ErrorBoundary>
    )
    expect(wrap).toMatchSnapshot()
  })

  it('getDerivedStateFromError should work correctly', function() {
    const error = { code: 404, message: 'Error' }
    const errorState = ErrorBoundary.getDerivedStateFromError(error)
    expect(errorState).toMatchSnapshot()
  })
})
