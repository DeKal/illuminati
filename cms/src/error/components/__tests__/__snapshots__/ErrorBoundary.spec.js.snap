// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary getDerivedStateFromError should work correctly 1`] = `
Object {
  "error": Object {
    "code": 404,
    "message": "Error",
  },
  "hasError": true,
}
`;

exports[`ErrorBoundary should render children content correctly 1`] = `
<div>
  Content
</div>
`;

exports[`ErrorBoundary should render error correctly 1`] = `
<Error
  error={
    Object {
      "msg": "Error",
    }
  }
/>
`;
